### PR TITLE
rework recipe to use actual cmake install

### LINKED ISSUE #12: Package doesn't include cmake configuration files
### Solution to issue cannot be found in the documentation.

- [X] I checked the documentation.

### Issue

The current recipe just copies the headers as opposed to doing a proper install which then creates and places cmake files so other libraries can use a `find_package` call as expected.

https://github.com/conda-forge/thrust-feedstock/blob/master/recipe/meta.yaml#L14

Coinciding cub issue: https://github.com/conda-forge/cub-feedstock/issues/16

### Installed packages

```shell
n/a
```

### Environment info

```shell
n/a
```

### PR DESCRIPTION
**Pull request by kkraus14**
Checklist
* [x] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [x] Bumped the build number (if the version is unchanged)
* [ ] Reset the build number to `0` (if the version changed)
* [x] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [ ] Ensured the license file is being packaged.

Fixes #12

### CODE REVIEW

**Comment by kkraus14** [comment]
@conda-forge-admin, please rerender

**Comment by kkraus14** [comment]
@conda-forge-admin, please rerender

**Comment by kkraus14** [comment]
This should be ready for review! Somehow looks like I got the selector incantation correct on the first try 😄

**Comment by kkraus14** [comment]
Nevermind, not sure why we're not getting ppc64le and aarch64 builds again 😞

**Comment by kkraus14** [comment]
@conda-forge-admin, please rerender

**Comment by kkraus14** [comment]
Well not sure if I handled the migrator right or not, but this should be ready for review.

File: recipe/meta.yaml [hunk 1]
```diff
@@ -10,17 +10,30 @@ source:
...
 build:
-  number: 0
-  script: mkdir -p "$PREFIX"/include && cp -r thrust "$PREFIX"/include  # [not win]
+  number: 1
+  skip: true  # [(linux or win) and (cuda_compiler_version != "11.2")]
+  ignore_run_exports_from:
+    - {{ compiler('cuda') }}  # [not osx]
+  script_env:
+    # for some reason /usr/local/cuda is not added to $PATH in the docker image
+    - CUDA_HOME  # [ppc64le or aarch64]
```
Comment (jakirkham) [inline recipe/meta.yaml:19]: xref: https://github.com/conda-forge/docker-images/issues/184

**Comment by jakirkham** [comment]
> Nevermind, not sure why we're not getting ppc64le and aarch64 builds again 😞

Migration for CUDA arch builds follows these steps ( https://github.com/conda-forge/conda-forge-pinning-feedstock/issues/2679#issuecomment-1084100146 ). The first step was done. Also the second step (CPU migration) was completed with PR ( https://github.com/conda-forge/thrust-feedstock/pull/13 ). However there wasn't actually any dependence on CUDA in this recipe before. So there was nothing to do for the third step and it was ignored entirely by this last migrator.

Now that this PR is adding CUDA dependence for this package CUDA arch migration will be tracked going forward. Though the migration will be marked completed when this is merged.

Side note: This can be seen a bit more clearly by noting arch CI builds that got removed in this PR. Those were for CPU only builds that are no longer relevant with the changes here.

> Well not sure if I handled the migrator right or not, but this should be ready for review.

Yeah it's just copying [this migrator file]( https://github.com/conda-forge/conda-forge-pinning-feedstock/blob/f36c7e06750cdb82a7abd25b676a9808f1f2c470/recipe/migrations/cuda_112_ppc64le_aarch64.yaml ) and placing it in `.ci_support/migrations/` then re-rendering.

**Comment by kkraus14** [comment]
Sounds like I did everything correctly then and this should be good to go.

**Review by rongou — APPROVED** [review]

**Review by leofang — APPROVED** [review]

**Comment by leofang** [comment]
Thanks, @kkraus14 @jakirkham!

**Pull request merged**